core-tests/shared/src/test/scala/zio/PromiseSpec.scala

```diff
@@ -1,165 +1,95 @@
+            v <- p.await
+          } yield assert(s, isTrue) && assert(v, equalTo(32))
+        },
+        testM("complete a promise using complete") {
+          for {
+            p <- Promise.make[Nothing, Int]
+            s <- p.complete(IO.succeed(14))
+            v <- p.await
+          } yield assert(s, isTrue) && assert(v, equalTo(14))
+        },
+        testM("fail a promise using fail") {
+          for {
+            p <- Promise.make[String, Int]
+            s <- p.fail("error with fail")
+            v <- p.await.run
+          } yield assert(s, isTrue) && assert(v, fails(equalTo("error with fail")))
+        },
+        testM("fail a promise using complete") {
+          for {
+            p <- Promise.make[String, Int]
+            s <- p.complete(IO.fail("error with done"))
+            v <- p.await.run
+          } yield assert(s, isTrue) && assert(v, fails(equalTo("error with done")))
+        },
+        testM("complete a promise twice") {
+          for {
+            p <- Promise.make[Nothing, Int]
+            _ <- p.succeed(1)
+            s <- p.complete(IO.succeed(9))
+            v <- p.await
+          } yield assert(s, isFalse) && assert(v, equalTo(1))
+        },
+        testM("interrupt a promise") {
+          for {
+            p <- Promise.make[Exception, Int]
+            s <- p.interrupt
+          } yield assert(s, isTrue)
+        },
+        testM("poll a promise that is not completed yet") {
+          for {
+            p       <- Promise.make[String, Int]
+            attempt <- p.poll
+          } yield assert(attempt, isNone)
+        },
+        testM("poll a promise that is completed") {
+          for {
+            p      <- Promise.make[String, Int]
+            _      <- p.succeed(12)
+            result <- p.poll.someOrFail("fail").flatten.run
+          } yield assert(result, succeeds(equalTo(12)))
+        },
+        testM("poll a promise that is failed") {
+          for {
+            p      <- Promise.make[String, Int]
+            _      <- p.fail("failure")
+            result <- p.poll.someOrFail("fail").flatten.run
+          } yield assert(result, fails(equalTo("failure")))
+        },
+        testM("poll a promise that is interrupted") {
+          for {
+            p      <- Promise.make[String, Int]
+            _      <- p.interrupt
+            result <- p.poll.someOrFail("fail").flatten.run
+          } yield assert(result, isInterrupted)
+        },
+        testM("isDone when a promise is completed") {
+          for {
+            p <- Promise.make[String, Int]
+            _ <- p.succeed(0)
+            d <- p.isDone
+          } yield assert(d, isTrue)
+        },
+        testM("isDone when a promise is failed") {
+          for {
+            p <- Promise.make[String, Int]
+            _ <- p.fail("failure")
+            d <- p.isDone
+          } yield assert(d, isTrue)
+        }
+      ),
+      timeout = TimeoutStrategy.Error(60.seconds)
```

Comment:
Stylistically, I would do `timeout(60.seconds) { ... suite ... }`

Reply:
@jdegoes @adamgfraser 
If we move it to `Runner`, that means we couldn't use `DefaultTestRunner` and `DefaultRunnableSpec` if we want to have a different one from the default. We would have to make a custom `Runner` and then a custom `RunnableSpec`, is that right?

Reply:
@jdegoes @ghostdogpr What if we made `TimeoutStrategy` a subtype of `TestAspect`? `TestAspect.timeout` is already isomorphic with `TestStrategy.Error`. `TestStrategy.Ignore` is a no-op, so that would be equivalent to `TestAspect.identity`, which we don't have but could easily add. So the only thing would be whether we could implement `Warn` as a `TestAspect`. I remember when I worked on the original PR I thought we couldn't for some reason but I think maybe I reworked it after that because the type signatures look the same. 

That would go a long way to unify things because then everything would be a `TestAspect` so at worst there are two different ways to add a `TestAspect` versus having this `TestAspect.timeout` and then `TimeoutStrategy` that are completely different things. I'm a little concerned with moving it to `Runner` because I think that makes it more difficult for us to have a default timeout policy that is also easily override by a user who is just extending `DefaultRunnableSpec` But maybe if we do that unification it addresses your concerns?